core/chains/cosmos.go

```diff
@@ -93,7 +154,38 @@ func fetch(url string) (cosmosResponse, error) {
 	var response cosmosResponse
 	err = json.Unmarshal(body, &response)
 	if err != nil {
-		return cosmosResponse{}, nil
+		return cosmosResponse{}, err
+	}
+
+	return response, nil
+}
+
+func fetchPool(url string) (cosmosPoolResponse, error) {
```

Comment:
add a comment for this function. what's a pool in the context of cosmos chains?

Reply:
I'll rename this as well as the validators one for uniformity.

Reply:
Further context:

`bonded_tokens` is just the direct way to get all of the valid (consensus/governance weight bearing) staked tokens, which is just a faster way to get the total amount than tallying all of the validators tokens
`not_bonded_tokens` is any that are either currently unstaking, or that are staked to inactive (`not_bonded`) validators.

```json
{
  "pool": {
    "not_bonded_tokens": "8646975757295",
    "bonded_tokens": "359227734437728"
  }
}
```